hermit-sys/src/net/mod.rs

```diff
@@ -426,19 +426,17 @@ pub fn sys_tcp_stream_read(handle: Handle, buffer: &mut [u8]) -> Result<usize, (
 			Ok(len) => {
 				return Ok(len);
 			}
-			Err(err) => {
-				match err {
-					ReadFailed::CanRecvFailed => {
-						// wait for tx buffers and try the send operation
-						if wait_for_result(handle, None, false) != WaitForResult::Ok {
-							return Ok(0);
-						}
-					}
-					_ => {
-						return Err(());
-					}
+			Err(ReadFailed::CanRecvFailed) => {
+				// wait for tx buffers and try the send operation
+				// ToDo: Is the != here correct? seems unintuitive to return ok if result is not okay
+				//	Additionally timeout of None seems like a bad idea
+				if wait_for_result(handle, None, false) != WaitForResult::Ok {
+					return Ok(0);
```

Comment:
Looks crazy, but I think that this is correct.  Maybe all buffers are used and the kernel is not able to receive any message. In this case, I give the control back to the application. At a later time, it could be possible to receive new data.

Reply:
Does it answer your question?

Reply:
Okay, that makes sense. Is there a specific reason why in that case `Ok(0)` is returned instead of `Err(())`?